SudokuProject/src/cspSolver/BTSolver.java

```diff
@@ -277,6 +278,59 @@ private Variable getMRV()
 	 */
 	private Variable getDegree()
 	{
+		List<Variable> tempNetwork = network.getVariables();
+		int maxDegree = 0;
+		Variable tempVar = tempNetwork.get(0);
+		for(Variable v: tempNetwork){
+			if(!v.isAssigned()){
+				int tempDegree = getDegree(v);
+				if(tempDegree > maxDegree){
+					maxDegree = tempDegree;
+					tempVar = v;
+				}
+			}
+		}
+		if(maxDegree != 0){
+			return tempVar;
+		}
+		return null;
+	}
+	/**
+	 *  This function gets the degree of a variable
+	 */
+	private int getDegree(Variable v){
+		int degree = 0;
+		for(Variable w: network.getNeighborsOfVariable(v)){
+			if(!w.isAssigned()){
+				degree++;
+			}
+		}
+		return degree;
+	}
+	/**
+	 * TODO: Implement MRV WITH DH
+	 * @return variable with minimum remaining values that isn't assigned, null if all variables are assigned DH as tie breaker.
+	 */
+	private Variable getMRVWithDH()
+	{
+		List<Variable> tempNetwork = network.getVariables();
+		int minLegalMoves = Integer.MAX_VALUE;
+		Variable tempVar = tempNetwork.get(0);
+		for(Variable v: tempNetwork){
+			if(!v.isAssigned())
+			{
+				if(v.size() == minLegalMoves && getDegree(v) > getDegree(tempVar)){

```

Comment:
We should keep track of the current degree of ```tempVar``` in a variable so we don't have to loop through all neighbors every time here.